tests/melt_compressible_advection.cc

```diff
@@ -0,0 +1,224 @@
+
+        values[0]=0;       //x vel
+        values[1]=0;    //y vel
+        values[2]=0;  // p_f
+        values[3]=0;  // p_c
+        values[4]=0;       //x melt vel
+        values[5]=0;    //y melt vel
+        values[6]=0;  // p_s
+        values[7]=0; // T
+        values[8]=porosity; // porosity
+      }
+  };
+
+
+  /**
+    * A postprocessor that evaluates the accuracy of the solution
+    * by using the L2 norm.
+    */
+  template <int dim>
+  class CompressibleMeltPostprocessor : public Postprocess::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+  {
+    public:
+      /**
+       * Generate graphical output from the current solution.
+       */
+      virtual
+      std::pair<std::string,std::string>
+      execute (TableHandler &statistics);
+  };
+
+  template <int dim>
+  std::pair<std::string,std::string>
+  CompressibleMeltPostprocessor<dim>::execute (TableHandler &statistics)
+  {
+    AssertThrow(Utilities::MPI::n_mpi_processes(this->get_mpi_communicator()) == 1,
+                ExcNotImplemented());
+
+    RefFunction<dim> ref_func;
+    const QGauss<dim> quadrature_formula (this->get_fe().base_element(this->introspection().base_elements.velocities).degree+2);
+
+    const unsigned int n_total_comp = this->introspection().n_components;
+    Vector<float> cellwise_errors_porosity (this->get_triangulation().n_active_cells());
+    ComponentSelectFunction<dim> comp_porosity(dim+2+dim+2, n_total_comp);
+
+    VectorTools::integrate_difference (this->get_mapping(),this->get_dof_handler(),
+                                       this->get_solution(),
+                                       ref_func,
+                                       cellwise_errors_porosity,
+                                       quadrature_formula,
+                                       VectorTools::L2_norm,
+                                       &comp_porosity);
+
+    std::ostringstream os;
+    os << std::scientific << cellwise_errors_porosity.l2_norm();
+    return std::make_pair("Error porosity_L2:", os.str());
+  }
+
+  template <int dim>
+  class PressureBdry:
+
+    public FluidPressureBoundaryConditions::Interface<dim>
+  {
+    public:
+      virtual
+      void fluid_pressure_gradient (
+        const MaterialModel::MaterialModelInputs<dim> &material_model_inputs,
+        const MaterialModel::MaterialModelOutputs<dim> &material_model_outputs,
+        std::vector<Tensor<1,dim> > &output
+      ) const
+      {
+        for (unsigned int q=0; q<output.size(); ++q)
+          {
+            const double y = material_model_inputs.position[q][1];
+            Tensor<1,dim> direction;
+            direction[dim-1] = 1.0;
+            output[q] = (1.1 + y)*direction;
+          }
+      }
+
+
+
+  };
+
+}
+
+// explicit instantiations
+namespace aspect
+{
+
+  ASPECT_REGISTER_MATERIAL_MODEL(CompressibleMeltMaterial,
+                                 "compressible melt material",
+                                 "")
```

Comment:
I know it is a test, but a short sentence about the material model? For the time being it is the only compressible one with melt? So people will try to understand it.